S7.Net/Tcp/ITcpClient.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace S7.Net.Tcp
+{
+    public interface ITcpClient
+    {
+        public int ReceiveTimeout { get; set; }
+
+        public int SendTimeout { get; set; }
+
+        public bool Connected { get; }
+
+        public void Close();
+
+        public Task ConnectAsync(string ip, int port, CancellationToken cancellationToken);
+
+        public Task ConnectAsync(string ip, int port);

```

Comment:
Can the CT have a default-value (here just the empty one), so one overload can be avoided?
```suggestion
        public Task ConnectAsync(string ip, int port, CancellationToken cancellationToken = default);

```